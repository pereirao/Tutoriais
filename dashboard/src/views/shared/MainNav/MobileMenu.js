// @flow

import React from 'react';
import classNames from 'classnames/bind';
import {connect} from 'react-redux';
import {push} from 'connected-react-router';
import I18n from '../../../localization';
import MBButton from '../../elements/MBButton';
import StoreRating from './StoreRating';
import {MAIN_NAV_ROUTE_NAMES} from './index';
import NotificationCount from './NotificationCount';
import style_sheet from './MainNav.module.css';


const cx = classNames.bind(style_sheet);

export const MobileMenu = ({hidden, current_path, pushRoute, close, toggleHelpModal}) => {
  if (hidden) return null;

  const handlePressLink = (route_name) => {
    if (!current_path.startsWith(`/${route_name}`)) {
      pushRoute(`/${route_name}`);
      close();
    }
  };

  return (
    <div className={style_sheet.mobileMenuWrapper}>
      <ul className={style_sheet.mobileMenuLinks}>
        {MAIN_NAV_ROUTE_NAMES.map(route_name => (
          <li
            key={`${route_name}link`}
            onClick={() => handlePressLink(route_name)}
            className={cx('mobileMenuLink', {activeLink: current_path.startsWith(`/${route_name}`)})}>
            {I18n.t(`ui.tab.${route_name}`)}
            {route_name === 'orders' && <NotificationCount/>}
          </li>
        ))}
      </ul>
      <div className={style_sheet.mobileMenuFooter}>
        <StoreRating/>
        <MBButton onClick={toggleHelpModal} button_type="link">
          <em>{I18n.t('ui.link.need_help')}</em>
        </MBButton>
      </div>
    </div>
  );
};

const MobileMenuDTP = {pushRoute: push};
const MobileMenuContainer = connect(null, MobileMenuDTP)(MobileMenu);

export default MobileMenuContainer;
