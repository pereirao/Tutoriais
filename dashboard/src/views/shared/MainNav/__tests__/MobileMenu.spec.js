import React from 'react';

// NOTE: named import is not connected to redux
import { MobileMenu } from '../MobileMenu';

it('renders MobileMenu correctly', () => {
  expect(shallow(<MobileMenu hidden={false} current_path="/" />)).toMatchSnapshot();
});
