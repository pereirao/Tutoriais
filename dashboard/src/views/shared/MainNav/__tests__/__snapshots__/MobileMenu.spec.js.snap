// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MobileMenu correctly 1`] = `
<div
  className="mobileMenuWrapper"
>
  <ul
    className="mobileMenuLinks"
  >
    <li
      className="mobileMenuLink"
      onClick={[Function]}
    >
      ORDERS
      <Connect(NotificationCount) />
    </li>
    <li
      className="mobileMenuLink"
      onClick={[Function]}
    >
      REPORTS
    </li>
    <li
      className="mobileMenuLink"
      onClick={[Function]}
    >
      INVENTORY
    </li>
    <li
      className="mobileMenuLink"
      onClick={[Function]}
    >
      SETTINGS
    </li>
  </ul>
  <div
    className="mobileMenuFooter"
  >
    <Connect(StoreRating) />
    <MBButton
      button_type="link"
    >
      <em>
        need help?
      </em>
    </MBButton>
  </div>
</div>
`;
